network/src/traits.rs

```diff
@@ -104,3 +105,80 @@ pub trait ReliableNetwork: BaseNetwork {
         handlers
     }
 }
+
+#[async_trait]
+pub trait UnreliableNetwork2<Message: Clone + Send + Sync> {
+    async fn unreliable_send(&mut self, peer: PublicKey, message: &Message) -> JoinHandle<()>;
+
+    /// Broadcasts a message to all `peers` passed as an argument.
+    /// The attempts to send individual messages are best effort and will not be retried.
+    async fn unreliable_broadcast(
+        &mut self,
+        peers: Vec<PublicKey>,
+        message: &Message,
+    ) -> Vec<JoinHandle<()>> {
+        let mut handlers = Vec::new();
+        for peer in peers {
```

Comment:
Would it be more idiomatically rust-y to do something functional here, like make this whole function body just

peers.into_iter().map(<call to unreliable_send>).collect()

Or is that too opaque or not possible for some reason? (asking mostly for my own rust edification). The nice aspect of it would be that it's just one line, and maybe faster if it enables pre-sizing the new vector correctly, etc.

Reply:
Unfortunately that doesn't work with the async call in the for loop otherwise i would prefer the method you've suggested.